### PR TITLE
Update dependant modules to Kotlin 1.4.30

### PR DESCRIPTION
**Pull request by antoinelamy**
## Description
- Migrate dependant modules to Kotlin to `1.4.30`
- Migrate dependant modules Gradle scripts to use Kotlin DSL over Groovy
- Bump major version as it contains a breaking change
- Bundle sources as part of aar binaries
- Use a local dependency over trikot.analytics instead of a published one

## Motivation and Context
Make Trikot available for Kotlin `1.4.30`

## How Has This Been Tested?
N/A

## Types of changes
Breaking change

### CODE REVIEW

**Review by marcantoinefortier — COMMENTED** [review]
👀

File: firebase-ktx/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+group = "com.mirego.trikot.analytics"
+
+configurations.forEach { it.exclude("org.reactivestreams") }
+
+dependencies {
+    api(project(":analytics"))
+    api("com.mirego.trikot:streams:${project.extra["trikot_streams_version"]}")
+    api("com.mirego.trikot:trikotFoundation:${project.extra["trikot_foundation_version"]}")
+    implementation("com.google.firebase:firebase-analytics:18.0.2")
+    implementation("androidx.lifecycle:lifecycle-extensions:2.2.0")
```
Comment (mathieularue) [inline firebase-ktx/build.gradle.kts:27]: `2.3.0`

File: firebase-ktx/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    implementation("androidx.lifecycle:lifecycle-reactivestreams-ktx:2.3.0")
+}
+
+android {
+    buildToolsVersion = "29.0.3"
```
Comment (mathieularue) [inline firebase-ktx/build.gradle.kts:32]: I don't think we need to specify a version of build tools anymore

File: mixpanel-ktx/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+group = "com.mirego.trikot.analytics"
+
+configurations.forEach { it.exclude("org.reactivestreams") }
+
+dependencies {
+    api(project(":analytics"))
+    api("com.mirego.trikot:streams:${project.extra["trikot_streams_version"]}")
+    api("com.mirego.trikot:trikotFoundation:${project.extra["trikot_foundation_version"]}")
+    implementation("com.mixpanel.android:mixpanel-android:5.8.2")
+    implementation("androidx.lifecycle:lifecycle-extensions:2.2.0")
```
Comment (mathieularue) [inline mixpanel-ktx/build.gradle.kts:27]: `2.3.0`

File: mixpanel-ktx/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    implementation("androidx.lifecycle:lifecycle-reactivestreams-ktx:2.3.0")
+}
+
+android {
+    buildToolsVersion = "29.0.3"
```
Comment (mathieularue) [inline mixpanel-ktx/build.gradle.kts:32]: Same

**Review by mathieularue — APPROVED** [review]

File: firebase-ktx/build.gradle.kts (hunk 1 above)
Comment (antoinelamy) [reply firebase-ktx/build.gradle.kts]: Removed, it was in fact unused

File: mixpanel-ktx/build.gradle.kts (hunk 3 above)
Comment (antoinelamy) [reply mixpanel-ktx/build.gradle.kts]: Also removed, unused

File: firebase-ktx/build.gradle.kts (hunk 2 above)
Comment (antoinelamy) [reply firebase-ktx/build.gradle.kts:31]: Removed

File: mixpanel-ktx/build.gradle.kts (hunk 4 above)
Comment (antoinelamy) [reply mixpanel-ktx/build.gradle.kts]: Removed

**Pull request merged**